### PR TITLE
#1830 add indicator dynamic column logic

### LINKED ISSUE #1830: Add indicator dynamic column logic
## Is your feature request related to a problem? Please describe.

Add actions/reducers for dynamically updating data table columns.

## Describe the solution you'd like

See above.

## Implementation

Add reducers to to `dataTableUtilities/pageReducers` following existing design. Update react-redux mappers if needed (`mapStateToProps`, `mapDispatchToProps`).

## Describe alternatives you've considered

See #1829.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #1830.

## Change summary

Adds dynamic column swapping logic to supplier requisition. Basic working idea, still may need some tidying up.

### Setup

- Create a new HIV indicator program.
- Add the program to the mobile store (i.e. set store tags, master list).
- Sync to mobile.

### Test cases

- [ ] Existing columns display correctly on supplier requisition initialisation.
- [ ] Indicator columns are displayed when indicators are toggled.
- [ ] When toggling between HIV, REGIMEN indicators, columns are dynamically updated. 

### Related areas to think about

Next steps are to implement cell actions/reducers for indicator value logic.

### CODE REVIEW

File: src/pages/dataTableUtilities/getPageInitialiser.js [hunk 1]
```diff
@@ -365,6 +365,18 @@ const supplierRequisitionInitialiser = requisition => {
       ? sortedData
       : sortedData.filter(item => item.isLessThanThresholdMOS);
 
+  const indicators = program?.indicators?.slice() || [];
+  const [selectedIndicator] = indicators;
```
Comment (joshxg) [inline src/pages/dataTableUtilities/getPageInitialiser.js]: this only works on actual arrays, I believe, whereas this *I think* is a `Realm.Results`, which is actually an object with the shape `{ 0, 1, 2 }`

File: src/pages/dataTableUtilities/getPageInitialiser.js [hunk 2]
```diff
@@ -365,6 +365,18 @@ const supplierRequisitionInitialiser = requisition => {
...
+
+  const indicatorColumns = selectedIndicator.columns.map(column => ({
```
Comment (joshxg) [inline src/pages/dataTableUtilities/getPageInitialiser.js]: `({valueType: type, description:key .....}) => ` etc looks neat

File: src/pages/dataTableUtilities/reducer/tableReducers.js [hunk 3]
```diff
@@ -144,6 +144,30 @@ export const toggleShowFinalised = state => {
...
+    type: column.valueType,
+    key: column.description,
+    title: column.description,
+    width: 10,
+    sortable: false,
+    editable: false,
+  }));
+
+  return { ...state, selectedIndicator, indicatorColumns };
+};
```
Comment (joshxg) [inline src/pages/dataTableUtilities/reducer/tableReducers.js]: second time, depending on if you believe in the principal of two or three, could extract to a utility method

File: src/pages/dataTableUtilities/getPageInitialiser.js [hunk 4]
```diff
@@ -365,6 +365,18 @@ const supplierRequisitionInitialiser = requisition => {
...
+    type: column.valueType,
+    key: column.description,
+    title: column.description,
+    width: 10,
```
Comment (joshxg) [inline src/pages/dataTableUtilities/getPageInitialiser.js]: this seems like an oopsie

File: src/pages/dataTableUtilities/getPageInitialiser.js [hunk 5]
```diff
@@ -365,6 +365,18 @@ const supplierRequisitionInitialiser = requisition => {
...
+    sortable: false,
+    editable: false,
```
Comment (joshxg) [inline src/pages/dataTableUtilities/getPageInitialiser.js]: possibly this too - is there a way to differentiate between the static, uneditable`IndicatorData` and the enterable?

File: src/pages/dataTableUtilities/getPageInitialiser.js (hunk 1 above)
Comment (wlthomson) [reply src/pages/dataTableUtilities/getPageInitialiser.js]: That's the idea for the `.slice()` here, I believe that returns a JS array?

Comment (joshxg) [reply src/pages/dataTableUtilities/getPageInitialiser.js]: haha whoops

Comment (wlthomson) [reply src/pages/dataTableUtilities/getPageInitialiser.js]: cookie

File: src/pages/dataTableUtilities/getPageInitialiser.js (hunk 2 above)
Comment (wlthomson) [reply src/pages/dataTableUtilities/getPageInitialiser.js]: b400a2b3f52cb01b278581614fc68b21601bedbb.

File: src/pages/dataTableUtilities/reducer/tableReducers.js (hunk 3 above)
Comment (wlthomson) [reply src/pages/dataTableUtilities/reducer/tableReducers.js]: 3fb6f83a5db933e2ed33e30c267cdb4e7afd8ee2.

File: src/pages/dataTableUtilities/getPageInitialiser.js (hunk 5 above)
Comment (wlthomson) [reply src/pages/dataTableUtilities/getPageInitialiser.js]: Just for now keeping these as constants, but have made a desktop issue related this.

**Pull request merged**